src/java/org/apache/nutch/indexer/IndexingJob.java

```diff
@@ -110,6 +134,27 @@ public void map(String key, WebPage page, Context context)
         }
         return;
       }
+      
+      if (deduplicate) {
+        Duplicate duplicate = duplicateStore.get(
```

Comment:
Hi @benmvachon, does these lines add a quadratic complexity on the number of duplicates? I mean if there are 10,000 URLs for a signature (maybe from "Internal Server Error"), for all 10,000 URLs the list of 10,000 duplicates is looped over to determine which URL is "original". This may cause that 100 million times a URL is parsed and split to count the number of path segments. Please correct me if I'm wrong. But if this is the case, wouldn't it be better to determine the "original" URL in the reducer of the DeduplicatorJob and save it in the duplicate store.

Reply:
@sebastian-nagel , you're right. I'll move duplicate filtering into the deduplictor reducer. Thank you